lib/matplotlib/widgets.py

```diff
@@ -2957,22 +2959,73 @@ def _release(self, event):
 
         self.update()
         self._active_handle = None
+        self._extents_on_press = None
 
         return False
 
     def _onmove(self, event):
         """Motion notify event handler."""
+
+        state = self._state | self._default_state
         # resize an existing shape
         if self._active_handle and self._active_handle != 'C':
             x0, x1, y0, y1 = self._extents_on_press
-            if self._active_handle in ['E', 'W'] + self._corner_order:
-                x1 = event.xdata
-            if self._active_handle in ['N', 'S'] + self._corner_order:
-                y1 = event.ydata
+            sizepress = [x1 - x0, y1 - y0]
```

Comment:
```suggestion
            size_on_press = [x1 - x0, y1 - y0]
```
I think changing this variable name makes it much clearer.